recipes/thrift/all/conanfile.py

```diff
@@ -0,0 +1,145 @@
+import os
+from conans import tools, CMake, ConanFile
+
+
+class ConanFileDefault(ConanFile):
+    name = "thrift"
+    description = "Thrift is an associated code generation mechanism for RPC"
+    topics = ("conan", "thrift", "serialization", "rpc")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/apache/thrift"
+    license = "Apache-2.0"
+    exports_sources = ["CMakeLists.txt"]
+    generators = "cmake", "cmake_find_package"
+
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_zlib": [True, False],
+        "with_libevent": [True, False],
+        "with_openssl": [True, False],
+        "with_boost_functional": [True, False],
+        "with_boost_smart_ptr": [True, False],
+        "with_boost_static": [True, False],
+        "with_boostthreads": [True, False],
+        "with_stdthreads": [True, False],
+        "with_c_glib": [True, False],
+        "with_cpp": [True, False],
+        "with_java": [True, False],
+        "with_python": [True, False],
+        "with_haskell": [True, False],
+        "with_plugin": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_zlib": True,
+        "with_libevent": True,
+        "with_openssl": True,
+        "with_boost_functional": False,
+        "with_boost_smart_ptr": False,
+        "with_boost_static": False,
+        "with_boostthreads": False,
+        "with_stdthreads": True,
+        "with_c_glib": False,
+        "with_cpp": True,
+        "with_java": False,
+        "with_python": False,
+        "with_haskell": False,
+        "with_plugin": False
+    }
+    _source_subfolder = "source_subfolder"
+    _build_subfolder = "build_subfolder"
+
+    _cmake = None
+
+    def config_options(self):
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = "thrift-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def build(self):
+        if self.options.with_openssl:
+            tools.replace_in_file("FindOpenSSL.cmake", "OpenSSL_INCLUDE_DIR", "OPENSSL_INCLUDE_DIR")
+            tools.replace_in_file("FindOpenSSL.cmake", "OpenSSL_LIBRARIES", "OPENSSL_LIBRARIES")
```

Comment:
I would prefer patching the project instead. We have the control over thrift 0.13.0. it will not change, but cmake_find_package is dynamic and can be changed

Reply:
What tool do you recommend to get all the libraries needed to link to opennssl ? Until now I used this ad hoc tool https://github.com/bincrafters/conan-qt/blob/testing/5.15.0/conanfile.py#L703-L718. Does conan have an equivalent feature independent of the generators ?

Reply:
`self.deps_cpp_info["openssl"].requires` so you can iterate and list their libraries `self.deps_cpp_info["openssl"].libs`

Reply:
Anyway, I saying to replace `OPENSSL_LIBRARIES` to `OpenSSL_LIBRARIES` in thrift files only. Creating a new patch

Reply:
I cannot find documentation of this `self.deps_cpp_info["foo"].requires`. all the .requires field I find are properties of cpp_info.components["bar"]. Is it an undocumented feature ? That said, I'm using `self.deps_cpp_info[p].public_deps` which happens to be undocumented too.
I'll create the patch,but I just want to say that it has the same brittleness as what is done currently: both solution presuppose that the generated file is called "FindOpenSSL.cmake"and that the variables are called "OpenSSL_LIBRARIES" and "OpenSSL_INCLUDE_DIR". Patching the sources is more maintenance in the future versions of thrift.

Reply:
Yes, the documentation is a bit outdated, related to deps_cpp_info.